### PR TITLE
Use raw location samples in telemtry traces

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Goal
Change the location samples type used in telemetry traces from enhanced to raw. 

### Implementation
I simply swapped the callbacks and added a test.

### CODE REVIEW

**Review by Guardiola31337 — APPROVED** [review]
Left a couple of minor comments (not blocking the PR though).

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/TelemetryLocationAndProgressDispatcherTest.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        ThreadController.getMainScopeAndRootJob().scope
+    )
+
+    @Test
+    fun ignoreEnhancedLocationUpdates() {
+        val enhancedLocation: Location = mockk()
+        locationAndProgressDispatcher.onEnhancedLocationChanged(enhancedLocation, mockk())
+
+        assertFalse(
+            "this job should not be completed",
```
[the remark this answers is not in the document]
Comment (LukasPaczos) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/TelemetryLocationAndProgressDispatcherTest.kt:29]: I like to add them to boolean assertions as they will normally fail with only "assertion failed" message, not with "value not true". This makes the initial report more readable imo.

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/TelemetryLocationAndProgressDispatcherTest.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    var coroutineRule = MainCoroutineRule()
+
+    private val locationAndProgressDispatcher = TelemetryLocationAndProgressDispatcher(
+        ThreadController.getMainScopeAndRootJob().scope
+    )
+
+    @Test
+    fun ignoreEnhancedLocationUpdates() {
+        val enhancedLocation: Location = mockk()
+        locationAndProgressDispatcher.onEnhancedLocationChanged(enhancedLocation, mockk())
```
[the remark this answers is not in the document]
Comment (LukasPaczos) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/TelemetryLocationAndProgressDispatcherTest.kt:26]: Just one new line, let's address as we go if needed.

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/TelemetryLocationAndProgressDispatcherTest.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Guardiola31337) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/TelemetryLocationAndProgressDispatcherTest.kt:29]: The test name should be descriptive enough so this extra message is not needed. We can even write sentences in Kotlin thanks to backticked method names. The sky is the limit 🚀

**Pull request merged**